recipes/bloomfiltertrie/popcnt.patch

```diff
@@ -0,0 +1,22 @@
+diff --git a/src/intersection.c b/src/intersection.c
+index 56599bb..443418e 100755
+--- a/src/intersection.c
++++ b/src/intersection.c
+@@ -137,7 +137,7 @@ uint32_t* intersection_uint32_SIMD(uint32_t* list_a, uint32_t* list_b) {
+         //[ copy out common elements
+         __m128i p = _mm_shuffle_epi8(v_a, shuffle_mask[mask]);
+         _mm_storeu_si128((__m128i*)&C[count], p);
+-        count += _mm_popcnt_u32(mask);
++        count += __builtin_popcountl(mask);
```

Comment:
Is this a behavioral change? The original code does a 32-bit pop-count, your change would make that 64-bit.

Reply:
Due to compatibility reasons, I don't think we can use SSE 4.2 yet; though, TBH, I'm not 100 % sure what the expected minimum capabilities/extensions we target are.
Here, I think you can just use `__builtin_popcount` (minus the `l`ong); that's also what GCC would do anyway if SSE >=4.1 and POPCNT were enabled: https://github.com/gcc-mirror/gcc/blob/gcc-4_8_5-release/gcc/config/i386/popcntintrin.h#L35.